File: backend/model/course.go
package model

import "time"

// Course 课程包
type Course struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	UserID      uint      `json:"user_id"`
	Title       string    `json:"title" gorm:"size:200;not null"`
	Description string    `json:"description" gorm:"size:500"`
	Tags        string    `json:"tags" gorm:"size:500;default:''"`
	IsPublic    bool      `json:"is_public" gorm:"default:false"`
	ItemCount   int       `json:"item_count" gorm:"default:0"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// CourseItem 课程条目（英语句子+翻译）
type CourseItem struct {
	ID              uint      `json:"id" gorm:"primaryKey"`
	CourseID        uint      `json:"course_id" gorm:"index;not null"`
	EnglishSentence string    `json:"english_sentence" gorm:"type:text;not null"`
	ChineseTranslation string `json:"chinese_translation" gorm:"type:text"`
	SortOrder       int       `json:"sort_order" gorm:"default:0"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

// CreateCourseRequest 创建课程包请求
type CreateCourseRequest struct {
	Title       string   `json:"title" binding:"required"`
	Description string   `json:"description"`
	Tags        []string `json:"tags"`
	IsPublic    bool     `json:"is_public"`
}

// UpdateCourseRequest 更新课程包请求
type UpdateCourseRequest struct {
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Tags        []string `json:"tags"`
	IsPublic    *bool    `json:"is_public"`
}

// CreateCourseItemRequest 创建课程条目请求
type CreateCourseItemRequest struct {
	EnglishSentence    string `json:"english_sentence" binding:"required"`
	ChineseTranslation string `json:"chinese_translation"`
	SortOrder          int    `json:"sort_order"`
}

// UpdateCourseItemRequest 更新课程条目请求
type UpdateCourseItemRequest struct {
	EnglishSentence    string `json:"english_sentence"`
	ChineseTranslation string `json:"chinese_translation"`
	SortOrder          *int   `json:"sort_order"`
}

// BatchCreateCourseItemsRequest 批量创建课程条目请求
type BatchCreateCourseItemsRequest struct {
	Items []CreateCourseItemRequest `json:"items" binding:"required,min=1"`
}

// BatchDeleteCourseItemsRequest 批量删除课程条目请求
type BatchDeleteCourseItemsRequest struct {
	Ids []uint `json:"ids" binding:"required,min=1"`
}

// CourseResponse 课程包响应（包含条目数量）
type CourseResponse struct {
	Course
	Items []CourseItem `json:"items,omitempty"`
}

File: frontend/src/service/api/course.ts
import { request } from '../request';

export interface Course {
  id: number;
  user_id: number;
  title: string;
  description: string;
  tags: string;
  is_public: boolean;
  item_count: number;
  created_at: string;
  updated_at: string;
}

export interface CourseItem {
  id: number;
  course_id: number;
  english_sentence: string;
  chinese_translation: string;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface CourseDetail extends Course {
  items: CourseItem[];
}

export interface CreateCourseParams {
  title: string;
  description?: string;
  tags?: string[];
  is_public?: boolean;
}

export interface UpdateCourseParams {
  title?: string;
  description?: string;
  tags?: string[];
  is_public?: boolean;
}

export interface CreateCourseItemParams {
  english_sentence: string;
  chinese_translation?: string;
  sort_order?: number;
}

export interface UpdateCourseItemParams {
  english_sentence?: string;
  chinese_translation?: string;
  sort_order?: number;
}

export interface CourseListParams {
  show_all?: boolean;
  keyword?: string;
  is_public?: boolean;
  tag?: string;
  page?: number;
  page_size?: number;
}

export interface CourseListResult {
  list: Course[];
  total: number;
  page: number;
  page_size: number;
}

export function fetchCourseList(params: CourseListParams = {}) {
  return request<CourseListResult>({
    url: '/api/courses',
    method: 'get',
    params
  });
}

export function fetchCourseDetail(id: number) {
  return request<CourseDetail>({
    url: `/api/courses/${id}`,
    method: 'get'
  });
}

export function fetchCreateCourse(data: CreateCourseParams) {
  return request<Course>({
    url: '/api/courses',
    method: 'post',
    data
  });
}

export function fetchUpdateCourse(id: number, data: UpdateCourseParams) {
  return request<Course>({
    url: `/api/courses/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteCourse(id: number) {
  return request({
    url: `/api/courses/${id}`,
    method: 'delete'
  });
}

export function fetchCourseItems(courseId: number) {
  return request<CourseItem[]>({
    url: `/api/courses/${courseId}/items`,
    method: 'get'
  });
}

export function fetchCreateCourseItem(courseId: number, data: CreateCourseItemParams) {
  return request<CourseItem>({
    url: `/api/courses/${courseId}/items`,
    method: 'post',
    data
  });
}

export function fetchBatchCreateCourseItems(courseId: number, items: CreateCourseItemParams[]) {
  return request<CourseItem[]>({
    url: `/api/courses/${courseId}/items/batch`,
    method: 'post',
    data: { items }
  });
}

export function fetchUpdateCourseItem(courseId: number, itemId: number, data: UpdateCourseItemParams) {
  return request<CourseItem>({
    url: `/api/courses/${courseId}/items/${itemId}`,
    method: 'put',
    data
  });
}

export function fetchDeleteCourseItem(courseId: number, itemId: number) {
  return request({
    url: `/api/courses/${courseId}/items/${itemId}`,
    method: 'delete'
  });
}

export function fetchBatchDeleteCourseItems(courseId: number, ids: number[]) {
  return request({
    url: `/api/courses/${courseId}/items/batch`,
    method: 'delete',
    data: { ids }
  });
}
